pkg/apps/watch-appstream.py

```diff
@@ -0,0 +1,253 @@
+import os
+import sys
+import time
+import traceback
+
+import gzip
+import xml.etree.ElementTree as ET
+import json
+
+lang = os.environ.get("LANGUAGE")
+
+def attr_lang(elt):
+    return elt.attrib.get('{http://www.w3.org/XML/1998/namespace}lang')
+
+def element(xml, tag):
+    if lang:
+        for elt in xml.iter(tag):
+            if attr_lang(elt) == lang:
+                return elt
+    return xml.find(tag)
+
+def element_value(xml, tag):
+    elt = element(xml, tag)
+    return elt.text if elt is not None else None
+
+def convert_description(xml, use_lang = True):
+    if xml is None:
+        return None
+
+    want_lang = lang if use_lang else None
+
+    # Only the following constructs are allowed, and they all appear
+    # at the top level:
+    #
+    # <p>text</p>
+    # <ul><li>text</li>...</ul>
+    # <ol><li>text</li>...</ol>
+
+    # A description can have 'lang' attributes both on the actual
+    # <description> element and on the contained <p>, <ul>, and <ol>
+    # elements, but probably not on the <li> elements.
+
+    def text(xml):
+        return " ".join(xml.itertext())
+
+    res = [ ]
+    for c in xml:
+        if attr_lang(c) != want_lang:
+            continue
+        if c.tag == 'p':
+            res.append(text(c))
+        elif c.tag == 'ul' or c.tag == 'ol':
+            res.append({ "tag": c.tag, "items": map(text, list(c.findall('li'))) })
+
+    # If we found nothing that matches lang, fall back to default
+    if lang is not None and len(res) == 0:
+        res = convert_description(xml, False)
+
+    return res
+
+def convert_cached_icon(dir, origin, xml):
+    if 'width' in xml.attrib and 'height' in xml.attrib:
```

Comment:
Nitpick of course, but please pick one style of quoting (single or double).

Reply:
I try to use double quotes for text-like strings like user-facing messages, and single quotes for code-like strings like dict keys.  But I fail frequently because I really don't care.

Which style should I be using?